core/src/main/java/org/opensearch/sql/analysis/NestedAnalyzer.java

```diff
@@ -63,6 +102,9 @@ public LogicalPlan visitFunction(Function node, AnalysisContext context) {
             "path", generatePath(nestedField.toString())
         );
       }
+
+      // NestedAnalyzer visits all functions in SELECT clause, creates logical plans for each and
+      // merges them. This is to avoid another merge rule in LogicalPlanOptimizer:create().
       if (child instanceof LogicalNested) {
```

Comment:
can this if statement and the if statement starting on line 73 be pulled out into a separate function that passes the middle argument of `LogicalNested` return?

Reply:
Yeah thanks, this is cleaner.